forc-pkg/src/pkg.rs

```diff
@@ -2437,15 +2532,12 @@ pub fn build(
         if let TreeType::Library { .. } = built_package.tree_type {
             lib_namespace_map.insert(node, namespace.into());
         }
-        last_pkg = Some(built_package);
         source_map.insert_dependency(manifest.dir());
+        standardize_json_abi_types(&mut built_package.json_abi_program);
+        built_packages.push((node, (built_package, source_map)));
```

Comment:
We should be careful not to collect all dependencies in memory unnecessarily too - we only want to collect the `outputs`.
```suggestion
        if outputs.contains(&node) {
            built_packages.push((node, (built_package, source_map)));
        }
```